db/hints/internal/hint_storage.cc

```diff
@@ -43,12 +43,14 @@ namespace fs = std::filesystem;
 namespace db::hints {
 namespace internal {
 
+namespace {
+
```

Comment:
> An anonymous namespace should be preferred wherever possible.

Why?

Reply:
That's a valid question. `static` functions and those wrapped in an anonymous namespace are equivalent ways to achieve the same, so these changes don't actually change anything. The only explanation I can provide is:
* I personally always prefer anonymous namespaces except for some niche cases, so this is my coding style preference (which is not relevant for non-personal projects of course)
* when adding new things, it's easy to forget about the `static`. But if you put something in an anonymous namespace that shouldn't be there, the linker will immediately start screaming (in most cases) because some symbol will be undefined. This solution seems less error-prone to me as a result.

If you think this commit should be skipped, I can drop it. But if it's fine and just the commit message should be better (as in provide some relevant justification of the changes), I can try to reword the second bullet from this message.

Thank you for stressing on this. I should think whether changes I make are really something that should be done or perhaps not.

Reply:
That's fine, I just wanted to know the reason. Clear descriptions in commits are also important because they convey your intent to reviewers and then they can verify whether what you wrote aligns with it or not.

I don't think our coding style disallows anonymous namespaces or encourages them (I couldn't find anything in coding-style.md), but I can see that we do use them in the code, so it's fine.